src/Paprika/Data/NibblePath.cs

```diff
@@ -105,19 +136,16 @@ public static NibblePath FromRawNibbles(ReadOnlySpan<byte> nibbles, Span<byte> w
     /// <summary>
     /// Reuses the memory of this nibble path moving it to odd position.
     /// </summary>
-    public NibblePath UnsafeMakeOdd(int odd)
+    public void UnsafeMakeOdd()
     {
-        if (odd == 0)
-            return this;
-
         Debug.Assert(_odd == 0, "Should not be applied to odd");
 
         for (int i = Length; i > 0; i--)
         {
             UnsafeSetAt(i, 0, GetAt(i - 1));
```

Comment:
I was thinking that this is the hot spot, especially for lengthy ones.

Reply:
Updated